js/src/controls/WidgetControl.js

```diff
@@ -0,0 +1,91 @@
+var widgets = require('@jupyter-widgets/base');
+var _ = require('underscore');
+var L = require('../leaflet.js');
+var control = require('./Control.js');
+var LeafletControlView = control.LeafletControlView;
+var LeafletControlModel = control.LeafletControlModel;
+
+L.Control.WidgetControl = L.Control.extend({
+    options: {
+        position: 'topleft',
+    },
+
+    _updateContent: function () {
+        if (!this._content) { return; }
+        this._container.appendChild(this._content);
+    },
+
+    update: function () {
+        if (!this._map) { return; }
+        this._updateContent();
+    },
+
+    getContent: function(){
+        return this._content;
+    },
+
+    setContent: function(content){
+        this._content = content;
+        this.update();
+        return this;
+    },
```

Comment:
One last thing ;)
You've got two unneeded methods `_updateContent` and `update`. Technically your code results in this:
```javascript
setContent: function(content){
    if (!this._map) { return; }
    this._content = content;
    this._container.appendChild(this._content);
    return this;
},
```